### PR TITLE
[DO NOT MERGE] Add timers on RowMerger to log RowMerger state when rows are not being emitted

### PR DESCRIPTION
**Pull request by vermas2012**
Updating the java source to 1.8

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable-hbase/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/RowMerger.java [hunk 1]
```diff
@@ -409,6 +436,43 @@ private FlatRow buildRow() {
...
+
+  /** Resets the timer for waiting on next row. Will be called when a row is emitted. */
+  private void rescheduleNextRowTask() {
+    if (nextRowFuture != null) {
+      nextRowFuture.cancel(false);
+    }
+    lastRowEmittedAtMillis = System.currentTimeMillis();
+    nextRowFuture =
+        NEXT_ROW_MONITOR_EXECUTOR_SERVICE.scheduleAtFixedRate(
+            () -> {
```
Comment (kolea2) [inline bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/RowMerger.java:450]: please switch this to `new Runnable(...` and revert the compile source change

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/RowMerger.java (hunk 1 above)
Comment (vermas2012) [reply bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/RowMerger.java]: done.

File: bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/RowMerger.java [hunk 2]
```diff
@@ -454,6 +522,10 @@ public final void onNext(ReadRowsResponse readRowsResponse) {
         }
 
         if (chunk.getCommitRow()) {
+          // No side effect of setting it every time. It just needs to turn true after first row is
+          // received.
+          recievedFirstRow = true;
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/RowMerger.java:527]: rec**ei**vedFirstRow

Comment (vermas2012) [reply bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/RowMerger.java]: done.

**Pull request closed without merging**